node/service/src/lib.rs

```diff
@@ -418,28 +502,89 @@ pub fn new_full<RuntimeApi, Executor>(
 		})
 		.collect();
 
-	let (overseer, overseer_handler) = real_overseer(leaves, prometheus_registry.as_ref(), spawner)?;
-	let overseer_handler_clone = overseer_handler.clone();
+	let authority_discovery_service = {
```

Comment:
We need to take an argument to `new_full`:

```
pub enum IsCollator {
    Yes,
    No
}
```

And enable authority discovery for collators as well.

Reply:
Would a solution to #1848 solve this for you, instead of adding an `IsCollator` enum?

Reply:
Is the collator an authority node? No, so this would not solve this problem.

Reply:
Also in the collator case we would need to run authority_discovery in `Role::Discover` mode.

Reply:
Then would it make sense to make `Collator` a `Role` variant?

Reply:
No that makes no sense. Otherwise I would have mentioned this. 

Authority discovery already has the special role `Discovery` that will be used, as already mentioned by @ordian.

Reply:
d1d4163